src/d2d_log.f90

```diff
@@ -131,6 +131,28 @@ module subroutine d2d_listing(given_io_unit)
 #endif
     write (io_unit, *) '==========================================================='
     write (io_unit, *) '==========================================================='
+#ifdef DEBUG
+    !
+    ! In DEBUG mode, rank 0 will also print environment variables
+    !
+    ! The system call, if writing to a file, is not blocking if supported
+    !
+    if (nrank == 0) then
+       write (io_unit, *) '============== Environment variables ======================'
+       write (io_unit, *) '==========================================================='
+       write (io_unit, *) '==========================================================='
+       if (io_unit == output_unit ) then
+          call execute_command_line("env", wait = .true.)
+       else
+          inquire(unit = io_unit, name = fname, iostat = ierror)
```

Comment:
Probably a dumb question - does this set fname? Otherwise it appears unset

Reply:
It should set fname. See the link below. This might not be bulletproof but it was working in my test.

[https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/file-operation-i-o-statements/inquire-statement-specifiers/inquire-name-specifier.html#inquire-name-specifier](https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/file-operation-i-o-statements/inquire-statement-specifiers/inquire-name-specifier.html#inquire-name-specifier)

Reply:
Marking the conversation as resolved. The option to add the name of the file as an argument remains open.